boards/CMakeLists.txt

```diff
@@ -26,7 +26,7 @@ target_compile_options(board_interface INTERFACE
     -ffunction-sections -fdata-sections -fmessage-length=0)
 target_link_options(board_interface INTERFACE
     -specs=nano.specs -specs=nosys.specs -Wl,--gc-sections
-    -Wl,--wrap=malloc -Wl,--wrap=free -Wl,--wrap=calloc)
+    -Wl,--wrap=malloc -Wl,--wrap=free -Wl,--wrap=calloc -Wl,--wrap=realloc)
```

Comment:
What's this? Are we trying to overwrite malloc(), free(), calloc() and realloc() here? But didn't you say we can't overwrite them?

Reply:
We can override those things with this compiler feature. Actually, only `malloc` and `free` are implemented. The `calloc` and `realloc` without implementation is to trigger a linker error when trying to link with some library that calls either of the two.